### DISCUSSION TITLE
changing the column type

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [apache/superset#18840] [Q&A / Help]
Hello,
Thanks for the awesome project.
you really democratizing Business Intelligence industry.
A question though, I have changed a column "price" from text to double precision. (inside legacy datasource editor)
![image](https://user-images.githubusercontent.com/32805117/155033121-93cbe252-9abe-424b-b53a-0a71e0739c4a.png)
However, it is not working as numeric column.
For example I can not make a filter with that column.
![image](https://user-images.githubusercontent.com/32805117/155033413-4982e485-ba61-4d32-b9fc-ed922926f583.png)
![image](https://user-images.githubusercontent.com/32805117/155033463-28c808f6-6aa3-4270-b28d-e673ffc72b24.png)
and also it behaves like a string.
here is sorted column.
![image](https://user-images.githubusercontent.com/32805117/155033809-20102966-bf73-4d66-a2ad-3a20e17f7394.png)

Is it a bug or I am doing something wrong?
Thanks for help

**Comment by villebro** [comment]
@Shahin-rmz what database are you using? Is `DOUBLE PRECISION` the native type in your database?

**Reply by Shahin-rmz** [reply]
I uploaded a csv file into a default ''example'' database, which is postgres.

**Comment by xavier-GitHub76** [comment]
Hello,
With the version 3.0.0, it is possible to set "Column Data Types"
#21906
